### PR TITLE
Initial packages

### PR DESCRIPTION
**Pull request by lawsonamzn**
This populates the first three initial packages and can serve as examples for further package development.

xxhash - pure source package (header only) committed as-is
OpenSSL - built using vcpkg
Lua - built using pull_and_build_from_git.py (included)

### CODE REVIEW

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+#
+
+
+set(CMAKE_SYSTEM_NAME iOS)
+set(CMAKE_SYSTEM_VERSION 1)
+set(CMAKE_OSX_ARCHITECTURES arm64)
+
+
+set(LY_IOS_CODE_SIGNING_IDENTITY "iPhone Developer" CACHE STRING "iPhone Developer")
```
Comment (spham-amzn) [inline Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: I'm not 100% certain, but all the PAL_ and LY_ attributes can be removed for 3rd party builds. This toolchain was copied from O3DE

**Review by amzn-changml — APPROVED** [review]

File: package_build_list_host_darwin.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "libsamplerate-0.2.1-rev2-ios": "package-system/libsamplerate-ios",
+        "OpenSSL-1.1.1b-rev2-mac": "package-system/OpenSSL-mac",
+        "OpenSSL-1.1.1b-rev2-ios": "package-system/OpenSSL-ios",
+        "tiff-4.2.0.10-mac" : "package-system/tiff-mac",
+        "tiff-4.2.0.10-ios" : "package-system/tiff-ios",
+        "python-3.7.10-rev1-darwin" : "python/darwin_x64/package",
+        "asn1-0.9.27-rev2-ios" : "package-system/asn1-ios",
+        "ASTCEncoder-2017_11_14-rev2-multiplatform" : "package-system/ASTCEncoder-multiplatform",
+        "PhysX-4.1.2.29882248-rev3-mac": "package-system/PhysX-mac",
+        "PhysX-4.1.2.29882248-rev3-ios": "package-system/PhysX-ios",
```
Comment (moraaar) [inline package_build_list_host_darwin.json:54]: This is not the latest from perforce. Please sync the latest. It's missing NvCloth, poly2tri and v-hacd libraries as build from source for darwin, linux and windows.

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake (hunk 1 above)
Comment (lawsonamzn) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: I'm not actually sure what you mean.  Isn't this used to make packages for ios?  It doesn't pull anything in from o3de.

File: package_build_list_host_darwin.json (hunk 2 above)
Comment (lawsonamzn) [reply package_build_list_host_darwin.json:54]: I intend to sync with latest after this initial commit.   I need to get a working system in first (this snapshot is from like, a week  or two ago and I know all the packages actually work with the scripts).

Once this is in it is my intention to then get latest from p4 and do another PR to get it up to date.

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: I think @spham-amzn means that the PAL_PLATFORM_NAME and PAL_PLATFORM_NAME_LOWERCASE is O3DE engine specific and is not needed for a general iOS build.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by spham-amzn — COMMENTED** [review]
For projects that are pulled in from different git repos, their cmake scripts will do nothing with the LY_ environments, so they wont be used.

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake (hunk 1 above)
Comment (spham-amzn) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: Yes, same goes with the LY_* variables

**Review by spham-amzn — APPROVED** [review]

**Review by moraaar — APPROVED** [review]

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake (hunk 1 above)
Comment (lawsonamzn) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: Do we have to remove the code signing then?
It has this
set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED YES)
set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${LY_IOS_CODE_SIGNING_IDENTITY})
set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM ${LY_IOS_DEVELOPMENT_TEAM})

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+include(${CMAKE_CURRENT_LIST_DIR}/SDK_ios.cmake)
+
+set(CMAKE_XCODE_ATTRIBUTE_TARGETED_DEVICE_FAMILY "1,2")
+set(CMAKE_XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET ${LY_IOS_DEPLOYMENT_TARGET})
+set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
+
+set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE NO)
+
+set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED YES)
```
[the remark this answers is not in the document]
Comment (lawsonamzn) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:31]: I'll remove all that stuff and try again.

File: Scripts/cmake/Platform/iOS/Toolchain_ios.cmake (hunk 1 above)
Comment (lawsonamzn) [reply Scripts/cmake/Platform/iOS/Toolchain_ios.cmake:14]: will remove and see if the current packages build

**Comment by lawsonamzn** [comment]
Note that I did re-test the one that builds with cmake (LUA) before pushing.

**Pull request merged**